modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayIcon from '@clayui/icon';
+import {ClayTooltipProvider} from '@clayui/tooltip';
+import classNames from 'classnames';
+import Proptypes from 'prop-types';
+import React from 'react';
+
+import {getDataRendererById} from '../utilities/dataRenderers';
+import {getSchemaString} from '../utilities/index';
+import DefaultRenderer from './DefaultRenderer';
+
+function TooltipTableRow({
+	contentRenderer,
+	divider,
+	highlighted,
+	itemData,
+	label,
+	value,
+	...options
+}) {
+	let DataRenderer = DefaultRenderer;
+
+	if (contentRenderer) {
+		DataRenderer = getDataRendererById(contentRenderer);
+	}
+
+	return (
+		<tr
+			className={classNames(
+				divider && 'with-divider',
+				highlighted && 'highlighted'
+			)}
+		>
+			<td className="table-cell-expand table-column-text-start">
+				{label}
+			</td>
+			<td className="table-column-text-end text-right">
+				<DataRenderer
+					itemData={itemData}
+					options={options}
+					value={value}
+				/>
+			</td>
+		</tr>
+	);
+}
+
+function TooltipResumeRenderer({itemData, options, value}) {
+	if (!value || !options.details?.rowsDefinitions) {
+		return null;
+	}
+
+	const tooltipTableRows = options.details.rowsDefinitions.reduce(
+		(rowsData, rowDefinition) => {
+			const value = getSchemaString(
+				itemData,
+				rowDefinition.valueFieldKey
+			);
+
+			if (value) {
+				return [
+					...rowsData,
+					{
+						...rowDefinition,
+						value,
+					},
+				];
```

Comment:
Once thing about this pattern is that you are copying the array on every step, producing a lot of short-lived intermediate objects that apply pressure on the garbage collector for no good reason. Seeing as you "own" the `rowsData` you can safely mutate it and avoid the copying. eg.

```js
const tooltipTableRows = options.details.rowsDefinitions.reduce(
	(rowsData, rowDefinition) => {
		const value = getSchemaString(itemData, rowDefinition.valueFieldKey);

		if (value) {
			rowsData.push({
				...rowDefinition,
				value,
			});
		}

		return rowsData;
	},
	[]
);
```